cmd/kubelet/app/options/options.go

```diff
@@ -464,8 +464,8 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 
 	fs.Int32Var(&c.RegistryPullQPS, "registry-qps", c.RegistryPullQPS, "If > 0, limit registry pull QPS to this value.  If 0, unlimited.")
 	fs.Int32Var(&c.RegistryBurst, "registry-burst", c.RegistryBurst, "Maximum size of a bursty pulls, temporarily allows pulls to burst to this number, while still not exceeding registry-qps. Only used if --registry-qps > 0")
-	fs.Int32Var(&c.EventRecordQPS, "event-qps", c.EventRecordQPS, "If > 0, limit event creations per second to this value. If 0, unlimited.")
-	fs.Int32Var(&c.EventBurst, "event-burst", c.EventBurst, "Maximum size of a bursty event records, temporarily allows event records to burst to this number, while still not exceeding event-qps. Only used if --event-qps > 0")
+	fs.Int32Var(&c.EventRecordQPS, "event-qps", c.EventRecordQPS, "QPS to limit event creations. If 0 will use DefaultQPS: 5.")
```

Comment:
The comment suggests we want to be able to set this to unlimited. Is this a bug that 0 does not? Will a negative number work? Do we want to fix the functionality to match the doc?

Reply:
https://github.com/kubernetes/kubernetes/blob/v1.20.2/pkg/kubelet/apis/config/validation/validation.go#L62-L64

a negative number doesn't work.

Reply:
> The comment suggests we want to be able to set this to unlimited. Is this a bug that 0 does not? Will a negative number work? Do we want to fix the functionality to match the doc?

https://github.com/kubernetes/kubernetes/blob/v1.20.2/pkg/kubelet/apis/config/validation/validation.go#L62-L64

a negative number doesn't work.

Reply:
https://github.com/kubernetes/kubernetes/blob/v1.20.2/cmd/kubelet/app/server.go#L593

But if we disable validation, a negative number will not enable RateLimiter, and it will be unlimited, heartbeatClient use -1 now.

Reply:
@ehashman should we disable validation and pass negative number to client-go?

Reply:
We shouldn't disable validation, but it may need an update. It seems like 0 was intended to set this to no limit, but in the case it's set to 0, the default is used instead.

/cc @pacoxu
who did some similar validation-related work in #96675

Reply:
```go
		if s.EventRecordQPS == 0 {
			eventClientConfig.QPS = float32(-1)
		} else {
			eventClientConfig.QPS = float32(s.EventRecordQPS)
		}
```
like this?

Reply:
```go
	if kc.EventRecordQPS < 0 {
		allErrors = append(allErrors, fmt.Errorf("invalid configuration: eventRecordQPS (--event-qps) %v must not be a negative number", kc.EventRecordQPS))
	}
```
or remove EventRecordQPS validation pass negative number.

Reply:
We can try to add the information about if 0 is used default will be used as well as the value shouldn't be negative

Reply:
add `The number must be >= 0.`

Reply:
Sorry for the delay.

The case is similar. 
0 always means default in the configuration.
nagative number means disable.